File: solutions/swift/difference-of-squares/1/Sources/DifferenceOfSquares/DifferenceOfSquares.swift
class Squares {
    let number: Int

    init(_ number: Int) {
        self.number = number
    }

    var sumOfSquares: Int {
        return (number * (number + 1) * (2 * number + 1)) / 6
    }

    var squareOfSum: Int {
        let sum = (number * (number + 1)) / 2
        return sum * sum
    }

    var differenceOfSquares: Int {
        return squareOfSum - sumOfSquares
    }
}

File: solutions/r/difference-of-squares/1/difference-of-squares.R

sum_of_squares <- function(number) {
  return ((number * (number + 1) * (2 * number + 1)) / 6)
}

square_of_sum <- function(number) {
  sum <- (number * (number + 1)) / 2
  return (sum * sum)
}

difference_of_squares <- function(number) {
  return (square_of_sum(number) - sum_of_squares(number))
}
